### PR TITLE
Chore(repo): Introduce pull request template

### PR DESCRIPTION
**Pull request by literat**
* @see: https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/creating-a-pull-request-template-for-your-repository

refs #DS-473, #383

Creating a PR template to help us write a better description for our PRs. :-)

### CODE REVIEW

**Review by tomassychra — APPROVED** [review]
👌

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -17,7 +17,7 @@ This project uses `makefile` for managing various chores, like dependency instal
 
 This project is a monorepo managed by [Lerna][lerna-home]. This means that each folder inside the [packages/][packages] directory represents a package published to npm. The root directory also contains _package.json_ file but this is only used for local development purposes and does not represent something that is ever published to npm (you might notice that it only contains _devDependencies_ and no prod deps 🤷‍♂️).
 
-## Commit messages
+## Commit Conventions
```
Comment (crishpeen) [inline CONTRIBUTING.md:20]: Change it in the content above too, please.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+<!-- e.g. is there anything you'd like reviewers to focus on? -->
+
+### Issue reference
+
+<!-- Please insert a link to the solved issue. If none, create one for this PR and then reference it here -->
+
+---
+
+### What is the purpose of this pull request? <!-- (put an "X" next to an item) -->
```
Comment (crishpeen) [inline .github/PULL_REQUEST_TEMPLATE.md:17]: Do we really need this? I thought this should be obvious from the commit prefix and/or the labels.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+<!-- Please insert your description here and provide especially info about the "what" this PR is solving -->
+
+### Additional context
+
+<!-- e.g. is there anything you'd like reviewers to focus on? -->
+
+### Issue reference
+
+<!-- Please insert a link to the solved issue. If none, create one for this PR and then reference it here -->
```
Comment (crishpeen) [inline .github/PULL_REQUEST_TEMPLATE.md:13]: Do we want to add links to our internal Jira? Or just add the IDs of the issues? I don't mind anything, I just try to not share links to internal software in public :)

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+---
+
+### What is the purpose of this pull request? <!-- (put an "X" next to an item) -->
+
+- [ ] Bug fix
+- [ ] New Feature
+- [ ] Documentation update
+- [ ] Other
+
+### Before submitting the PR, please make sure you do the following
```
Comment (crishpeen) [inline .github/PULL_REQUEST_TEMPLATE.md:24]: This means that I will have to check all these checkboxes now when creating a new PR all the time? Is this a way to stop me or slow me down to not make that many PRs? :D

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 3 above)
Comment (literat) [reply .github/PULL_REQUEST_TEMPLATE.md:13]: I know, but this is happening right now. So I want to make this clear and discuss this if we do this or not.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (literat) [reply .github/PULL_REQUEST_TEMPLATE.md:17]: You are right, this duplicates that thing.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 4 above)
Comment (literat) [reply .github/PULL_REQUEST_TEMPLATE.md:24]: I am not quite sure yet. But this should help us to follow our guidelines and double-check and fill in the information needed. We both know that we are submitting PRs without proper descriptions. I know I do that :-)

But if this makes no sense to you we can modify this. I also think about guidelines of healthy metrics that should be here.

File: CONTRIBUTING.md (hunk 1 above)
Comment (literat) [reply CONTRIBUTING.md:20]: Fixed.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 3 above)
Comment (literat) [reply .github/PULL_REQUEST_TEMPLATE.md:13]: We are allowed to do this. So I am letting this here.

**Comment by literat** [comment]
Ok, the best CR will be the usage itself. :D Merging.

**Pull request merged**